R/reshape_output.R

```diff
@@ -0,0 +1,257 @@
+    age = numeric(),
+    # The length associated with the parameter or derived quantity
+    length = numeric(),
+    # The modeled time perioed that the value pertains to
+    time = integer(),
+    # The initial value use to start the optimization procedure
+    initial = numeric(),
+    # The estaimted parameter value, which would be the MLE estimate or the value
+    # used for a given MCMC iteration
+    estimate = numeric(),
+    # Estimated uncertainty, reported as a standard deviation
+    uncertainty = numeric(),
+    # The pointwise log-likelihood used for the estimation model
+    log_like = numeric(),
+    # The pointwise log-likelihood used for the test or holdout data
+    log_like_cv = numeric(),
+    # The gradient component for that parameter, NA for derived quantities
+    gradient = numeric(),
+    # A TRUE/FALSE indicator of whether the parameter was estimated (and not fixed),
+    # with NA for derived quantities
+    estimated = logical()
+  )
+  
+  if (length(sdreport) > 0) {
+    std <- summary(sdreport)
+    # Number of rows for derived quantities: based on the difference
+    # between the total number of rows in std and the length of parameter_names.
+    derived_quantity_nrow <- nrow(std) - length(parameter_names)
+    # Create a tibble with the data from the std, and then apply transformations.
+    estimates <- estimates_outline |>
+      tibble::add_row(
+        label = dimnames(std)[[1]],
+        estimate = std[, "Estimate"],
+        uncertainty = std[, "Std. Error"],
+        # Use obj[["env"]][["parameters"]][["p"]] as this will return both initial
+        # fixed and random effects while obj[["par"]] only returns initial fixed
+        # effects
+        initial = c(
+          obj[["env"]][["parameters"]][["p"]],
+          rep(NA_real_, derived_quantity_nrow)
+        ),
+        gradient = c(
+          obj[["gr"]](opt[["par"]]),
+          rep(NA_real_, derived_quantity_nrow)
+        ),
+        estimated = c(
+          rep(TRUE, length(parameter_names)),
+          rep(NA, derived_quantity_nrow)
+        )
+      )
+  } else {
+    estimates <- estimates_outline |>
+      tibble::add_row(
+        label = names(obj[["par"]]),
+        initial = obj[["env"]][["parameters"]][["p"]],
+        estimate = obj[["env"]][["parameters"]][["p"]],
+        estimated = FALSE
+      )
+  }
+
+  estimates <- estimates |>
+    # Split labels and extract module, id, label, and parameter id
+    dplyr::mutate(label_splits = strsplit(label, split = "\\.")) |>
+    dplyr::rowwise() |>
+    dplyr::mutate(
+      module_name = ifelse(length(label_splits) > 1, label_splits[[1]], NA_character_),
+      module_id = ifelse(length(label_splits) > 1, as.integer(label_splits[[3]]), NA_integer_),
+      label = ifelse(length(label_splits) > 1, label_splits[[2]], label),
+      parameter_id = ifelse(length(label_splits) > 1, as.integer(label_splits[[4]]), NA_integer_)
+    ) |>
```

Comment:
Just a note that this section of code might be easier with `tidyr::separate_wider_*()` functions.